wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -541,10 +556,12 @@ data Transaction = Transaction
     -- ^ The input money distribution.
   , txOutputs       :: !(NonEmpty PaymentDistribution)
     -- ^ The output money distribution.
-  , txType          :: TransactionType
+  , txType          :: !TransactionType
     -- ^ The type for this transaction (e.g local, foreign, etc).
-  , txDirection     :: TransactionDirection
+  , txDirection     :: !TransactionDirection
     -- ^ The direction for this transaction (e.g incoming, outgoing).
+  , txCreationTime  :: !(V1 Core.Timestamp)
```

Comment:
I like having microsecond precision here. However, a unix timestamp is an awfully non-human-friendly way to specify this -- using a UTCTime would be more "human-readable" while offering the same level of precision. On the other hand, a UTCTime is a lot less friendly to manipulate for computers, and has to deal with the vagaries of leap seconds etc.

Reply:
FWIW UTCTime can represent sub-second resolutions.

A datatype like

```haskell
data ApiTime = UTC UTCTime | Timestamp Timestamp
```

with a non-tagged parse attempt would be fine, IMO.